src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -485,17 +485,15 @@ private bool IsAssemblyOrClassCleanupMethod(MethodInfo methodInfo, Type cleanupA
             TestClassInfo classInfo,
             ref MethodInfo[] initAndCleanupMethods)
         {
-            if (initAndCleanupMethods is null)
+            if (initAndCleanupMethods.Any(x => x != null))
```

Comment:
If you run the included tests but without the fixes you can see the problems that are being solved. Because this method used to reset `initAndCleanupMethods` back to null, it would cause a NullReferenceException elsewhere in the loop. So rather than the array being null, just the elements are null in this new process

Reply:
Yes what Matt says is correct. This is the main fix for the null ref issue.